test/cuda.jl

```diff
@@ -24,6 +24,11 @@ end
   g_gpu = gradient(x -> w(x), a_gpu)[1]
   @test g_gpu isa CuArray
   @test g_gpu |> collect ≈ g
+
+  # https://github.com/FluxML/Zygote.jl/issues/1027
+  @test gradient(x -> sum(x .!= 0), a_gpu) == (nothing,)
+  g3 = gradient(x -> sum(x .^ 3) ./ count(x .> 3), a)[1]
+  @test cu(g3) ≈ gradient(x -> sum(x .^ 3) ./ sum(x .> 3), a_gpu)[1]
```

Comment:
This new test causes "ERROR: a exception was thrown during kernel execution." https://buildkite.com/julialang/zygote-dot-jl/builds/515

Reply:
The last line `gradient(x -> sum(x .^ 3) ./ sum(x .> 3), a_gpu)` still gives a kernel error, I have no idea why. The pieces are fine separately, `sum(x .^ 3)` and `sum(x .> 3)` (and now tested). 

It's not a regression, fails with a similar error on Zygote v0.6.14, CUDA v3.3.0.

Reply:
GPU failure on Julia 1.5 is this, surely unrelated:
```
Removing coverage files
--
  | Running Pkg.gc()
  | ERROR: type Context has no field verbose
  | Stacktrace:
  | [1] #Context!#44 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Types.jl:384 [inlined]
  | [2] gc(::Pkg.Types.Context; collect_delay::Day, kwargs::Base.Iterators.Pairs{Symbol,Bool,Tuple{Symbol},NamedTuple{(:verbose,),Tuple{Bool}}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/Pkg/src/API.jl:341
  | [3] top-level scope at none:1
  | 🚨 Error: Error tearing down bootstrap: The plugin julia pre-exit hook exited with status 1
```